talk/expert/cpp20concepts.tex

```diff
@@ -0,0 +1,194 @@
+\subsection[tmpl]{C++20 Requirements and Concepts}
+
+\begin{frame}[fragile]
+  \frametitlecpp[20]{C++20 Requirements and Concepts}
+  \begin{block}{Motivation}
+    Generic programming is made of functions and class templates
+    which can be instantiated with different types. It is frequent
+    to instantiate them with {\em irrelevant types}, and the resulting
+    compilation errors are generally very long and hardly understandable.
+    As a last resort, the template authors are providing {\em documentation}
+    about the relevant parameters, and practice some tricky
+    {\em template meta-programmation}. C++20 finally brings simpler ways to
+    define constraint on template parameters ! Among different proposals,
+    the ISO committee has validated the flavor known as {\em Concepts Lite}.
+  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[20]{C++20 Requirements and Concepts}
+  \begin{block}{Requirements and concepts in a nutshell}
+    \begin{itemize}
+    \item A template can define requirements on some of its type parameters.
+    \item Compiler error messages clearly state which parameter value does not fulfill which expected requirement.
+    \item A typical set of requirements can be gathered in a reusable concept.
+    \item Overload resolution takes those requirements and concepts into account.
+    \item The standard library now provides many concepts, easy to use.
+    \item Writing a new perfect concept stays an expert topic.
+    \end{itemize}
+  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[20]{C++20 Requirements and Concepts}
+  \begin{block}{Old-style SFINAE example}
+    The old way to forbid some type arguments is the definition of
+    fake template arguments, which lead to a substitution failure when
+    the type does not fulfill the condition.
```

Comment:
Are you explaining SFINAE somewhere? Because if not, this is really too short to explain it. At least you should mention that SFINAE removes overload form an overload set. If you want to explain SFINAE at all.